### PR TITLE
Synchronization extension implementation

### PR DESCRIPTION
**Pull request by bserdar**

### CODE REVIEW

File: crud/src/test/java/com/redhat/lightblue/crud/mongo/AbstractMongoCrudTest.java [hunk 1]
```diff
@@ -79,8 +80,8 @@ public void setup() throws Exception {
 
     @After
     public void teardown() throws Exception {
-        mongo.getConnection().dropDatabase(DATABADSE_NAME);
-        coll = null;
+        if(coll!=null)
+            coll.remove(new BasicDBObject());
```
Comment (dcrissman) [inline crud/src/test/java/com/redhat/lightblue/crud/mongo/AbstractMongoCrudTest.java]: Not sure I understand this change. Why is this better?

Comment (bserdar) [inline crud/src/test/java/com/redhat/lightblue/crud/mongo/AbstractMongoCrudTest.java]: Dropping the database is done asynchronously. So after dropDatabase
returns, the next test starts running, and then the actual drop takes
place, screwing it up.

Removing eveything is synchronous.

Comment (dcrissman) [inline crud/src/test/java/com/redhat/lightblue/crud/mongo/AbstractMongoCrudTest.java]: Do we still want to set coll = null on the way out?

Comment (bserdar) [inline crud/src/test/java/com/redhat/lightblue/crud/mongo/AbstractMongoCrudTest.java]: No need. Collection still exists.

File: crud/src/main/java/com/redhat/lightblue/crud/mongo/MongoCRUDController.java [hunk 2]
```diff
@@ -366,6 +380,14 @@ public void updatePredefinedFields(CRUDOperationContext ctx, JsonDoc doc) {
     }
 
     @Override
+    public Extension getExtensionInstance(Class<? extends Extension> extensionClass) {
+        if(extensionClass.equals(LockingSupport.class))
+            return new MongoLockingSupport(this);
+        else
+            return null;
```
Comment (dcrissman) [inline crud/src/main/java/com/redhat/lightblue/crud/mongo/MongoCRUDController.java]: Should it throw an exception, or maybe log a warning?

Comment (bserdar) [inline crud/src/main/java/com/redhat/lightblue/crud/mongo/MongoCRUDController.java]: No, API spec says if it doesn't support it, it returns null.

Not all CRUD implementations will support all extensions.

File: crud/src/test/java/com/redhat/lightblue/crud/mongo/AbstractMongoCrudTest.java (hunk 1 above)
Comment (dcrissman) [inline crud/src/test/java/com/redhat/lightblue/crud/mongo/AbstractMongoCrudTest.java]: Ok, I am good with that then.

**Comment by dcrissman** [comment]
Moving the configuration files will likely cause problems for test cases in downstream projects.

**Comment by dcrissman** [comment]
I created PRs for the downstream projects that I found that need new config files. I will go ahead and merge this PR and it is easy enough to see if anything was missed.

**Pull request merged**